### PR TITLE
A series of minor validator test changes

### PR DESCRIPTION
**Pull request by kmaglione**
See linked bugs for details, if necessary.

### CODE REVIEW

File: validator/testcases/javascript/call_definitions.py [hunk 1]
```diff
@@ -167,6 +167,26 @@ def spellcheck_updatecurrentdictionary(wrapper, arguments, traverser):
...
 
 
+def webbrowserpersist(wrapper, arguments, traverser):
+    """
+    Most nsIWebBrowserPersist should no longer be used, in favor of the new
+    Downloads.jsm interfaces.
+    """
+    traverser.err.warning(
+        err_id=("testcases_javascript_call_definititions",
+         "webbrowserpersist"),
```
Comment (magopian) [inline validator/testcases/javascript/call_definitions.py]: Strange indendation, pretty sure flake8 will complain

File: validator/testcases/javascript/predefinedentities.py [hunk 2]
```diff
@@ -267,7 +300,12 @@
                         e.get_resource("em:bootstrap") and
                         "Authors of bootstrapped add-ons must take care "
                         "to remove any added observers "
-                        "at shutdown."}}},
+                        "at shutdown."}},
+         "dangerous": lambda a, t, e:
+            lambda t, e: (e.metadata.get("is_jetpack") and
+                "The observer service should not be used directly in SDK "
```
Comment (magopian) [inline validator/testcases/javascript/predefinedentities.py]: Is flake8 happy with this indentation?

File: validator/testcases/regex.py [hunk 3]
```diff
@@ -334,6 +362,23 @@ def rewrap():
...
 @register_generator
+class ExtensionManagerRegexTests(RegexTestGenerator):
+    """
+    Tests for uses of the old extension manager API, which should not be
+    referenced in new extensions.
+    """
+
+    def js_tests(self):
+        yield self.get_test(
+                r"@mozilla\.org/extensions/manager;1|"
```
Comment (magopian) [inline validator/testcases/regex.py]: To much indentation

**Comment by magopian** [comment]
I don't have the knowledge regarding the rules/warnings/tests added here, who does and could review this?

Also @kmaglione please rebase this branch now that I've merged some of your other PRs, thanks!

File: validator/testcases/javascript/call_definitions.py (hunk 1 above)
Comment (kmaglione) [inline validator/testcases/javascript/call_definitions.py]: Yeah. I copied that from from the test below and decided to just err on the side of consistency, but I'll just change them both. (flake8 incidentally complains so much about this file that looking at the output is barely even useful)

File: validator/testcases/javascript/predefinedentities.py (hunk 2 above)
Comment (kmaglione) [inline validator/testcases/javascript/predefinedentities.py]: Nope.

File: validator/testcases/regex.py (hunk 3 above)
Comment (kmaglione) [inline validator/testcases/regex.py]: Hm. This is the indentation used for every other test in this file (except for a couple of others that I added in this PR). I think they should all be consistent, so should I leave these as-is, or fix all the others?

**Comment by kmaglione** [comment]
I don't think there's currently anyone who's especially qualified to review the code at the moment. @jvillalobos can review the semantics, I suppose.

**Comment by kmaglione** [comment]
Rebased and fixed some of the indentation problems. I'd accidentally added the synchronous SQL tests both here and in #256, so I had to merge the functionality of the two.

File: validator/testcases/regex.py (hunk 3 above)
Comment (magopian) [inline validator/testcases/regex.py]: Yeah, I guess leave it like that, at least for now, maybe one day i'll do a "flake8 pass" on all of amo-validator. Thanks @kmaglione

**Comment by magopian** [comment]
Thanks a lot for all this work @kmaglione

**Pull request merged**